src/http_request.cpp

```diff
@@ -104,35 +104,63 @@ const http::header_view_map http_request::get_cookies() const {
     return get_headerlike_values(MHD_COOKIE_KIND);
 }
 
-std::string_view http_request::get_arg(std::string_view key) const {
-    std::map<std::string, std::string>::const_iterator it = cache->unescaped_args.find(std::string(key));
+void http_request::populate_args() const {
+    if (!cache->unescaped_args.empty()) {
+        return;
+    }
+    arguments_accumulator aa;
+    aa.unescaper = unescaper;
+    aa.arguments = &cache->unescaped_args;
+    MHD_get_connection_values(underlying_connection, MHD_GET_ARGUMENT_KIND, &build_request_args, reinterpret_cast<void*>(&aa));
+}
+
+http::http_arg_value http_request::get_arg(std::string_view key) const {
+    populate_args();
 
+    auto it = cache->unescaped_args.find(std::string(key));
```

Comment:
can we have a std::string_view comparator in http::arg_comparator to avoid this allocation?

Reply:
The arg_comparator supports string_view, but std::map and friends do not support heterogeneous lookup? Switching the unescaped args to be keyed by string_view breaks the file handling, because the keys end up spread across several requests and become invalidated.

Reply:
Since C++14 std::map supports lookup by an arbitrary key type `K` in `find()`

Reply:
Just had to rejig the comparator a bit. Thanks!